File: merge-sorted-array/merge-sorted-array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        i = 0
        j = 0
        ans = [0] * (m + n)
        while True:
            if i == m and j == n:
                break
            elif i == m and j != n:
                ans[i + j:] = nums2[j:n]
                break
            elif i != m and j == n:
                ans[i + j:] = nums1[i:m]
                break
            else:
                num1 = nums1[i]
                num2 = nums2[j]
                if num1 < num2:
                    ans[i + j] = num1
                    i += 1
                elif num1 > num2:
                    ans[i + j] = num2
                    j += 1
                else:
                    ans[i + j] = num1
                    ans[i + j + 1] = num2
                    i += 1
                    j += 1
        nums1[:] = ans[:]
        return nums1
        

File: merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] arr = new int[m + n];
        
        int aidx = 0;
        int bidx = 0;
        for(int i = 0; i < m + n; i++) {
            // need finish condition
            if(aidx == m) {
                arr[i] = nums2[bidx];
                bidx++;
                continue;
            }
            if(bidx == n) {
                arr[i] = nums1[aidx];
                aidx++;
                continue;
            }
            if(nums1[aidx] < nums2[bidx]) {
                arr[i] = nums1[aidx];
                aidx++;
            } else {
                arr[i] = nums2[bidx];
                bidx++;
            }
        }
        for(int i = 0; i < m + n; i++) {
            nums1[i] = arr[i];
        }
    }
}